src/utils/templating.rs

```diff
@@ -0,0 +1,52 @@
+use std::collections::HashMap;
+
+use bcrypt::DEFAULT_COST;
+use rand::distributions::{Alphanumeric, DistString};
+use tera::{Context, Function, Tera, Value};
+
+use crate::constants::PASSWORD_LEN;
+
+pub fn render(content: &str, parameters: &HashMap<String, String>) -> Result<String, tera::Error> {
+    // Create templating context
+    let mut context = Context::new();
+
+    // Fill context with parameters
+    for (name, value) in parameters {
+        context.insert(name, value)
+    }
+
+    // Create render engine
+    let mut tera = Tera::default();
+    tera.register_function("random_password", random_password());
+    tera.register_function("bcrypt", bcrypt());
+
+    // Render template
+    tera.render_str(content, &context)
+}
+
+fn random_password() -> impl Function {
+    Box::new(
```

Comment:
No need to box this, as far as I can tell from the signatures?

Reply:
I followed the official docs [here](https://tera.netlify.app/docs/#functions).

Reply:
Odd, [the API docs](https://docs.rs/tera/latest/tera/trait.Function.html#implementors) only mention a generic impl over `Fn`.

Reply:
(Which should still work with the boxing, but only because `Box<T>: Fn where T: Fn`.

Reply:
Odd indeed. I now removed the `Box` in f33a89f